src/test/java/Domain/Users/AssociationRepresentativeTest.java

```diff
@@ -126,12 +125,112 @@ public void removeReferee3ITest() {
         aR = new AssociationRepresentative(aRUser);
         SystemUser newRefereeUser = new SystemUser("refUsername", "refName");
         new Referee(newRefereeUser, "refTraining");
+
+        Referee referee = (Referee) newRefereeUser.getRole(RoleTypes.REFEREE);
+        Season season = new Season(new League("noName"),"2020/21");
+        season.assignReferee(referee);
+        referee.assignToSeason(season);
+
+        assertTrue(season.doesContainsReferee(referee));
         assertTrue(aR.removeReferee(newRefereeUser));
+        assertFalse(season.doesContainsReferee(referee));
+    }
+
+    @Test
+    public void assignRefereeToSeasonUTest() {
+        SystemUser aRUser = new SystemUser("arUsername", "arName");
+        aR = new AssociationRepresentative(aRUser);
+        SystemUser newRefereeUser = new SystemUserStub("refUsername", "refName", 93121);
+        Referee refereeRole = (Referee) newRefereeUser.getRole(RoleTypes.REFEREE);
+        //new RefereeStub(newRefereeUser, "refTraining");
+        Season season = new Season(new League("noName"), "2020/21");
+
+        boolean thrown = false;
+        try {
+            aR.assignRefereeToSeason(season, refereeRole);
+        } catch (Exception e) {
+            thrown = true;
+        }
+        assertFalse(thrown);
+
+        assertTrue(season.doesContainsReferee(refereeRole));
+        assertTrue(refereeRole.getSeasons().contains(season));
+    }
+
+    @Test
+    public void assignRefereeToSeason2UTest() {
+        SystemUser aRUser = new SystemUser("arUsername", "arName");
+        aR = new AssociationRepresentative(aRUser);
+        SystemUser newRefereeUser = new SystemUserStub("refUsername", "refName", 93121);
+        Referee refereeRole = (Referee) newRefereeUser.getRole(RoleTypes.REFEREE);
+        //new RefereeStub(newRefereeUser, "refTraining");
+        Season season = new Season(new League("noName"), "2020/21");
```

Comment:
This is Unit test but you are not using the stubs of some of the classes. If there is a reason, please explain when we can decide to not use stubs in Utests.

Reply:
צודק, שיניתי את השם.
זה התחיל כבדיקת יחידה והתפתח לאינטגרציה כדי להצליח להגיע לקטע הקוד הנדרש.